tests/mcp_server.py

```diff
@@ -1,19 +1,42 @@
-from mcp.server.fastmcp import FastMCP
+import asyncio
 
-mcp = FastMCP('PydanticAI MCP Server')
+from mcp.server import Server
+from mcp.server.stdio import stdio_server
+from mcp.types import TextContent, Tool
 
+server = Server('test-server')
+log_level = 'unset'
 
-@mcp.tool()
-async def celsius_to_fahrenheit(celsius: float) -> float:
-    """Convert Celsius to Fahrenheit.
 
-    Args:
-        celsius: Temperature in Celsius
+@server.call_tool()
+async def query_db(name: str, arguments: dict[str, float]) -> list[TextContent]:
+    if name == 'celsius_to_fahrenheit':
+        celsius = arguments['celsius']
+        return [TextContent(type='text', text=str((celsius * 9 / 5) + 32))]
+    elif name == 'get_log_level':
+        return [TextContent(type='text', text=log_level)]
```

Comment:
@Kludex I would like to emit a log here, even if it's not picked up by the client right now, but I couldn't find how to do it with the server.

Reply:
here's what i did in a logfire test:

```python
    fastmcp = FastMCP()

    @fastmcp.tool()
    async def random_number(ctx: Context) -> int:  # type: ignore
        await ctx.info('Generating a random number')
        await ctx.log(
            'alert',  # type: ignore  # mcp type hints problem
            'Dice broken! Improvising...',
            logger_name='my_logger',
        )
        return 4
```

Reply:
why did you switch away from fastmcp?